analyzers/tests/SonarAnalyzer.UnitTest/TestCases/PrivateFieldUsedAsLocalVariable.cs

```diff
@@ -214,6 +214,30 @@ protected string SomeString
         {
             get { return F37 ?? (F37 = "5"); }
         }
+
+        private int F38 = 0;
```

Comment:
Unfortunately, each time we need to update this rule, we need to spend lots of time understanding these test cases. I think we can improve this a lot by adding comments and by renaming the methods and fields to something which reflects the case tested instead of numbers.

We can start with this new case you add it and in time maybe update the full file.